### PR TITLE
MediaAttachment: Cleanup

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
- Updated MediaAttachmentDelegate's method signature. Swift 3 style!
- Updated few method's signature(s). Again, Swift 3 style.
- Moved Nested Types to an extension
- Updated visibility modifiers
- Wired guard let when possible

Ref. #682

### To test:
- Verify the unit tests are still green!
- Run a smoke test over Media. 

**Note:** No logic was changed, just code reorganization, and few minor touches!
cc @diegoreymendez

### CODE REVIEW

File: Aztec/Classes/TextKit/MediaAttachment.swift [hunk 1]
```diff
@@ -125,11 +104,20 @@ open class MediaAttachment: NSTextAttachment {
...
 
+    /// Attachment's Delegate
+    ///
     weak var delegate: MediaAttachmentDelegate?
-    
-    var isFetchingImage: Bool = false
+
+    /// Indicates if there's a download OP in progress, or not.
+    ///
+    fileprivate var isFetchingImage = false
```
Comment (diegoreymendez) [inline Aztec/Classes/TextKit/MediaAttachment.swift]: Are you sure this is not used by third part apps?  The main one being WPiOS.

File: Aztec/Classes/TextKit/MediaAttachment.swift [hunk 2]
```diff
@@ -227,18 +216,18 @@ open class MediaAttachment: NSTextAttachment {
...
-    func mediaBounds(forBounds bounds: CGRect) -> CGRect {
+    func mediaBounds(for bounds: CGRect) -> CGRect {
         let containerWidth = bounds.size.width
         let origin = CGPoint(x: xPosition(forContainerWidth: bounds.size.width), y: imageMargin)
         let size = CGSize(width: onScreenWidth(containerWidth), height: onScreenHeight(containerWidth) - imageMargin)
         return CGRect(origin: origin, size: size)
     }
 
-    fileprivate func glyph(basedOnImage image:UIImage, forBounds bounds: CGRect) -> UIImage? {
+    private func glyph(basedOnImage image:UIImage, forBounds bounds: CGRect) -> UIImage? {
```
Comment (diegoreymendez) [inline Aztec/Classes/TextKit/MediaAttachment.swift]: To stay consistent `basedOnImage` -> `for`, and probably `forBounds` -> `in` or `constrainedTo`.

File: Aztec/Classes/TextKit/MediaAttachment.swift (hunk 1 above)
Comment (jleandroperez) [inline Aztec/Classes/TextKit/MediaAttachment.swift]: Confirmed to be safe! WPiOS isn't hooked up to that flag (which should always have been internal!)

**Comment by jleandroperez** [comment]
@diegoreymendez comments addressed, ready for another look. Thank you!!!

**Comment by jleandroperez** [comment]
Thanks Diego!!

**Pull request merged**